canaille: 0.0.57 -> 0.0.74

erictapen opened the pull request:
[Changelog](https://gitlab.com/yaal/canaille/-/blob/0.0.74/CHANGES.rst)

Closes https://github.com/NixOS/nixpkgs/pull/395671

@azmeuk

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

erictapen committed:
canaille: 0.0.57 -> 0.0.74

Changelog: https://gitlab.com/yaal/canaille/-/blob/0.0.74/CHANGES.rst

erictapen committed:
python3Packages.flask-alembic: init at 3.1.1

erictapen committed:
python3Packages.pytest-scim2-server: init at 0.1.2

r-ryantm committed:
python312Packages.authlib: 1.5.1 -> 1.5.2

eljamm reviewed (approved):
Diff LGTM and VM tests are passing. I'm also running [nixpkgs-review-gha](https://github.com/eljamm/nixpkgs-review-gha/actions/runs/14730392399), but that might take a while given the number of rebuilds.

flokli reviewed (commented):
Please check for downstream rebuild success. https://github.com/NixOS/nixpkgs/pull/402797 showed the authlib bump also broke the synapse build.

erictapen commented:
@flokli Yeah I was wondering wether it might be worth it to merge authlib and just fix forward from there? From first glance it looks like quite the effort to fix everything before merging.

flokli commented:
> @flokli Yeah I was wondering wether it might be worth it to merge authlib and just fix forward from there? From first glance it looks like quite the effort to fix everything before merging.

No, this means breaking a lot of things, while we're somewhat close to the branchoff. The fixes should cook here in this PR.

erictapen committed:
fittrackee: 0.9.4 -> 0.9.8

Diff: https://github.com/SamR1/FitTrackee/compare/refs/tags/v0.9.4...v0.9.8

Changelog: https://github.com/SamR1/FitTrackee/blob/v0.9.8/CHANGELOG.md

erictapen commented:
Pinging matrix-synapse maintainers then: @sumnerevans @Ma27 @fadenb @mguentner @NickCao @teutat3s

Ma27 commented:
Can I have more context, please?
The authlib stuff needs fixing on synapse side?

flokli commented:
`matrix-synapse` build fails with the following:

```
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/build/source/tests/rest/client/test_login.py", line 1316, in test_login_aud_no_config
    self.assertEqual(
  File "/nix/store/pj3pzycszazvw73v66a2yq3k11hld846-python3.12-twisted-24.11.0/lib/python3.12/site-packages/twisted/trial/_synctest.py", line 444, in assertEqual
    super().assertEqual(first, second, msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 885, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 1251, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: "JWT validation failed: invalid_claim: Invalid claim 'aud'" != 'JWT validation failed: invalid_claim: Invalid claim "aud"'
- JWT validation failed: invalid_claim: Invalid claim 'aud'
?                                                     ^   ^
+ JWT validation failed: invalid_claim: Invalid claim "aud"
?                                                     ^   ^


tests.rest.client.test_login.JWTTestCase.test_login_aud_no_config
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/build/source/tests/rest/client/test_login.py", line 1297, in test_login_aud
    self.assertEqual(
  File "/nix/store/pj3pzycszazvw73v66a2yq3k11hld846-python3.12-twisted-24.11.0/lib/python3.12/site-packages/twisted/trial/_synctest.py", line 444, in assertEqual
    super().assertEqual(first, second, msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 885, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 1251, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: "JWT validation failed: invalid_claim: Invalid claim 'aud'" != 'JWT validation failed: invalid_claim: Invalid claim "aud"'
- JWT validation failed: invalid_claim: Invalid claim 'aud'
?                                                     ^   ^
+ JWT validation failed: invalid_claim: Invalid claim "aud"
?                                                     ^   ^


tests.rest.client.test_login.JWTTestCase.test_login_aud
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/build/source/tests/rest/client/test_login.py", line 1265, in test_login_iss
    self.assertEqual(
  File "/nix/store/pj3pzycszazvw73v66a2yq3k11hld846-python3.12-twisted-24.11.0/lib/python3.12/site-packages/twisted/trial/_synctest.py", line 444, in assertEqual
    super().assertEqual(first, second, msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 885, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 1251, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: "JWT validation failed: invalid_claim: Invalid claim 'iss'" != 'JWT validation failed: invalid_claim: Invalid claim "iss"'
- JWT validation failed: invalid_claim: Invalid claim 'iss'
?                                                     ^   ^
+ JWT validation failed: invalid_claim: Invalid claim "iss"
?                                                     ^   ^


tests.rest.client.test_login.JWTTestCase.test_login_iss
-------------------------------------------------------------------------------
Ran 4015 tests in 111.460s

FAILED (skips=166, failures=3, successes=3846)
```

I think it's a quotation difference in the error error messages returned by `authlib`, and the `matrix-synapse` test suite hardcoding one of them. This probably should be changed to not directly compare strings upstream, I assume other distros will run into this as well once bumping authlib.

erictapen commented:
Yes exactly, that's just what fixed it for canaille at least.

flokli committed:
matrix-synapse-unwrapped: fix compatibility with authlib-1.5.2

flokli commented:
I just rebased and pushed a commit fixing matrix-synapse-unwrapped, fetching the test suite fix sent over at https://github.com/element-hq/synapse/pull/18390.

teutat3s commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 402797`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 10 packages marked as broken and skipped:</summary>
  <ul>
    <li>chatgpt-retrieval-plugin</li>
    <li>chatgpt-retrieval-plugin.dist</li>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>python312Packages.weaviate-client</li>
    <li>python312Packages.weaviate-client.dist</li>
    <li>python313Packages.flask-alembic</li>
    <li>python313Packages.flask-alembic.dist</li>
    <li>python313Packages.weaviate-client</li>
    <li>python313Packages.weaviate-client.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 72 packages built:</summary>
  <ul>
    <li>canaille</li>
    <li>canaille.dist</li>
    <li>fittrackee</li>
    <li>fittrackee.dist</li>
    <li>home-assistant-component-tests.discovergy</li>
    <li>home-assistant-component-tests.home_connect</li>
    <li>home-assistant-component-tests.senz</li>
    <li>home-assistant-component-tests.vicare</li>
    <li>matrix-synapse</li>
    <li>matrix-synapse-plugins.matrix-synapse-ldap3</li>
    <li>matrix-synapse-plugins.matrix-synapse-ldap3.dist</li>
    <li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam</li>
    <li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam.dist</li>
    <li>matrix-synapse-plugins.matrix-synapse-s3-storage-provider</li>
    <li>matrix-synapse-plugins.matrix-synapse-s3-storage-provider.dist</li>
    <li>matrix-synapse-plugins.matrix-synapse-shared-secret-auth</li>
    <li>matrix-synapse-plugins.matrix-synapse-shared-secret-auth.dist</li>
    <li>matrix-synapse-plugins.synapse-http-antispam</li>
    <li>matrix-synapse-plugins.synapse-http-antispam.dist</li>
    <li>matrix-synapse-unwrapped</li>
    <li>matrix-synapse-unwrapped.dist</li>
    <li>pgadmin4</li>
    <li>pgadmin4-desktopmode</li>
    <li>pgadmin4-desktopmode.dist</li>
    <li>pgadmin4.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>powerdns-admin</li>
    <li>python312Packages.aiohomeconnect</li>
    <li>python312Packages.aiohomeconnect.dist</li>
    <li>python312Packages.aiosenz</li>
    <li>python312Packages.aiosenz.dist</li>
    <li>python312Packages.authlib</li>
    <li>python312Packages.authlib.dist</li>
    <li>python312Packages.flask-alembic</li>
    <li>python312Packages.flask-alembic.dist</li>
    <li>python312Packages.flask-security</li>
    <li>python312Packages.flask-security.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.kanidm</li>
    <li>python312Packages.kanidm.dist</li>
    <li>python312Packages.pydiscovergy</li>
    <li>python312Packages.pydiscovergy.dist</li>
    <li>python312Packages.pytest-scim2-server</li>
    <li>python312Packages.pytest-scim2-server.dist</li>
    <li>python312Packages.pyvicare</li>
    <li>python312Packages.pyvicare.dist</li>
    <li>safety-cli (python312Packages.safety)</li>
    <li>safety-cli.dist (python312Packages.safety.dist)</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.aiohomeconnect</li>
    <li>python313Packages.aiohomeconnect.dist</li>
    <li>python313Packages.aiosenz</li>
    <li>python313Packages.aiosenz.dist</li>
    <li>python313Packages.authlib</li>
    <li>python313Packages.authlib.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.kanidm</li>
    <li>python313Packages.kanidm.dist</li>
    <li>python313Packages.pydiscovergy</li>
    <li>python313Packages.pydiscovergy.dist</li>
    <li>python313Packages.pytest-scim2-server</li>
    <li>python313Packages.pytest-scim2-server.dist</li>
    <li>python313Packages.pyvicare</li>
    <li>python313Packages.pyvicare.dist</li>
    <li>python313Packages.safety</li>
    <li>python313Packages.safety.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>

flokli commented:
flask-alembic doesn't work with 3.13 yet, but we only need 3.12 for canaille (for now). I openend https://github.com/pallets-eco/flask-alembic/issues/47 and marked it as broken for 3.13.

I was able to reproduce the `weaviate-client` build failures on my Asahi MacBook:

```
pytest flags: -m pytest -k not\ \(test_bearer_token\)\ and\ not\ \(test_auth_header_with_catchall_proxy\)\ and\ not\ \(test_token_refresh_timeout\)\ and\ not\ \(test_with_simple_auth_no_oidc_via_api_key\)\ and\ not\ \(test_client_with_extra_options\) test mock_tests
/nix/store/bpbyc25gjaiid47a6l0s2mna582mjngv-python3.12-pytest-asyncio-0.26.0/lib/python3.12/site-packages/pytest_asyncio/plugin.py:217: PytestDeprecationWarning: The configuration option "asyncio_default_fixture_loop_scope" is unset.
The event loop scope for asynchronous fixtures will default to the fixture caching scope. Future versions of pytest-asyncio will default the loop scope for asynchronous fixtures to function scope. Set the default fixture loop scope explicitly in order to avoid unexpected behavior in the future. Valid fixture loop scopes are: "function", "class", "module", "package", "session"

  warnings.warn(PytestDeprecationWarning(_DEFAULT_FIXTURE_LOOP_SCOPE_UNSET))
============================= test session starts ==============================
platform linux -- Python 3.12.9, pytest-8.3.5, pluggy-1.5.0
rootdir: /build/source
configfile: pytest.ini
plugins: pytest_httpserver-1.1.2, asyncio-0.26.0, anyio-4.9.0, flaky-3.8.1, hypothesis-6.130.5, Faker-37.1.0, time-machine-2.16.0
asyncio: mode=Mode.STRICT, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collecting ... Fatal Python error: Aborted

Current thread 0x0000fffff7fb4e80 (most recent call first):
  File "<frozen importlib._bootstrap>", line 488 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 1293 in create_module
  File "<frozen importlib._bootstrap>", line 813 in module_from_spec
  File "<frozen importlib._bootstrap>", line 921 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "/nix/store/k1ii2djdn9gajqdvi8g01s5riildb4yq-python3.12-polars-1.27.1/lib/python3.12/site-packages/polars/__init__.py", line 24 in <module>
  File "<frozen importlib._bootstrap>", line 488 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 999 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "/build/source/test/collection/test_validator.py", line 5 in <module>
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/assertion/rewrite.py", line 185 in exec_module
  File "<frozen importlib._bootstrap>", line 935 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1331 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1360 in _find_and_load
  File "<frozen importlib._bootstrap>", line 1387 in _gcd_import
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/importlib/__init__.py", line 90 in import_module
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/pathlib.py", line 587 in import_path
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/python.py", line 493 in importtestmodule
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/python.py", line 546 in _getobj
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/python.py", line 284 in obj
  File "/nix/store/bpbyc25gjaiid47a6l0s2mna582mjngv-python3.12-pytest-asyncio-0.26.0/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 754 in _patched_collect
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/runner.py", line 389 in collect
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/runner.py", line 341 in from_call
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/runner.py", line 391 in pytest_make_collect_report
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/runner.py", line 567 in collect_one_node
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 835 in _collect_one_node
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 970 in genitems
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 975 in genitems
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 975 in genitems
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 809 in perform_collect
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 347 in pytest_collection
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 336 in _main
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/zqprw0jpynlji2251rndhp60ql0yimkz-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/nix/store/vcll1kfpf7gyzzcqbg8068awdw0gbgc6-python3.12-pytest-8.3.5/lib/python3.12/site-packages/pytest/__main__.py", line 9 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main

Extension modules: markupsafe._speedups, _time_machine, grpc._cython.cygrpc, google._upb._message, _cffi_backend, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing (total: 56)
/nix/store/8sz8w1lx8plk82cii8w785z3r8in08xr-pytest-check-hook/nix-support/setup-hook: line 23:   894 Aborted                 (core dumped) /nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/bin/python3.12 "${flagsArray[@]}"
```

I cannot reproduce them with x86_64-linux, and cannot reproduce them on a aarch64-linux with 4K pagesize, so I assume it's some pagesize-related issue, but definitely out of scope for this PR.

So nixpkgs-review on a 4K aarch64-linux machine or x86_64-linux machine should come back clean, and this is good to merge.
